addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/handler/VeraSceneHandler.java

```diff
@@ -0,0 +1,380 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.vera2.VeraBindingConstants;
+import org.openhab.binding.vera2.config.VeraSceneConfiguration;
+import org.openhab.binding.vera2.controller.json.Scene;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VeraSceneHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dmitriy Ponomarev
+ */
+public class VeraSceneHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private ScenePolling scenePolling;
+    private ScheduledFuture<?> pollingJob;
+    private VeraSceneConfiguration mConfig = null;
+
+    public VeraSceneHandler(Thing thing) {
+        super(thing);
+        scenePolling = new ScenePolling();
+    }
+
+    private class Initializer implements Runnable {
+        @Override
+        public void run() {
+            try {
+                VeraBridgeHandler veraBridgeHandler = getVeraBridgeHandler();
+                if (veraBridgeHandler != null && veraBridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                    ThingStatusInfo statusInfo = veraBridgeHandler.getThing().getStatusInfo();
+                    logger.debug("Change scene status to bridge status: {}", statusInfo);
+
+                    // Set thing status to bridge status
+                    updateStatus(statusInfo.getStatus(), statusInfo.getStatusDetail(), statusInfo.getDescription());
+
+                    try {
+                        logger.debug("Add channels");
+                        Scene scene = veraBridgeHandler.getController().getScene(mConfig.getSceneId());
+                        if (scene != null) {
+                            logger.debug("Finded {} scene", scene.name);
+                            addSceneAsChannel(scene);
+                        }
+                    } catch (Exception e) {

```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception